### PR TITLE
Separate ImmutableList<T> nested classes into different files

### PR DESCRIPTION
**Pull request by jamesqo**
This is part of my old PR https://github.com/dotnet/corefx/pull/15382, which was rejected way back in January because it had too many changes. This is just the part of that PR that separates `ImmutableList<T>` into different files; after this is merged, I'll port the rest of the changes over.

### CODE REVIEW

File: src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList_1.cs [hunk 1]
```diff
@@ -16,7 +14,7 @@ namespace System.Collections.Immutable
     /// </summary>
     /// <typeparam name="T">The type of elements in the set.</typeparam>
     [DebuggerDisplay("Count = {Count}")]
-    [DebuggerTypeProxy(typeof(ImmutableListDebuggerProxy<>))]
+    [DebuggerTypeProxy(typeof(ImmutableEnumerableDebuggerProxy<>))]
```
Comment (jamesqo) [inline src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList_1.cs]: @AArnott This proxy is being used with all of the other immutable collections already, and those were validated to work by @stephentoub at https://github.com/dotnet/corefx/pull/16008#issuecomment-279889732. Plus I think @JonHanna has some corefx tests in place for the collection proxies. I think we can safely assume the proxy won't malfunction.

FWIW, I tried building this branch and copying the S.C.I dll into my app today, but I ran into a stubborn problem with binding redirects I've been stuck on for a couple of hours. I'm throwing in the towel a bit here.

Comment (jamesqo) [inline src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList_1.cs]: @AArnott I finally managed to get a local build of S.C.I.dll running with a console app. Here's visual proof it works:

imlist_debug

**Comment by safern** [comment]
Thanks @jamesqo for taking care of this cleanup and for the visual proof.

**Pull request merged**